pkg/controller/network_policy.go

```diff
@@ -558,13 +647,74 @@ func (c *Controller) fetchPolicySelectedAddresses(namespace, protocol string, np
 			for _, podIP := range pod.Status.PodIPs {
 				if podIP.IP != "" && util.CheckProtocol(podIP.IP) == protocol {
 					selectedAddresses = append(selectedAddresses, podIP.IP)
+
+					svcs, err := c.servicesLister.Services(pod.Namespace).List(labels.Everything())
+					if err != nil {
+						klog.Errorf("failed to list svc, %v", err)
+						return nil, nil, fmt.Errorf("failed to list svc, %v", err)
+					}
```

Comment:
We can move this part to outside of the Pods loop.

Reply:
> We can move this part to outside of the Pods loop.

ok